app/services/bulk_update/recommendations_uploads/validate_csv.rb

```diff
@@ -10,26 +10,44 @@ def initialize(csv:, record:)
 
       # do all required headers exist in the CSV headers
       def validate!
-        required_headers
+        identifier_header!
+        date_header!
+        dates!
       end
 
     private
 
       attr_reader :csv, :record
 
-      VALID_HEADERS_SET = ::Set[
-        "trn",
-        "date qts or eyts standards met",
-      ].freeze
+      def identifier_header!
+        return if headers & identifying_headers != []
 
-      def required_headers
-        unless csv_headers_set.superset?(VALID_HEADERS_SET)
-          record.errors.add(:base, "CSV is missing the required headers")
-        end
+        record.errors.add(:file, :no_id_header)
       end
 
-      def csv_headers_set
-        @csv_headers_set ||= ::Set.new(csv.headers)
+      def identifying_headers
+        Reports::BulkRecommendReport::IDENTIFIERS.map(&:downcase)
+      end
+
+      def date_header!
+        return if headers.include?(date_header)
+
+        record.errors.add(:file, :no_date_header)
+      end
+
+      def date_header
+        Reports::BulkRecommendReport::DATE.downcase
+      end
+
+      def dates!
+        return unless date_header!
```

Comment:
I'm not sure this is working - if the date header is present, `date_header!` returns `nil` (see line 33). So if never continues to validate the presence of any dates.

We probably need to flesh out the tests in `validate_csv_spec` to include testing:

- `given a CSV with missing date header`
- `given a CSV with missing dates`

Reply:
separately, i didn't realise we were going to be validating "no dates" as a CSV level header, and in fact we already have content for the next page if someone uploads a CSV with no dates.

Which do we want to keep? If it's a CSV level validation then we need to delete `app/views/bulk_update/recommendations_uploads/_summary_no_dates.html.erb` and remove the logic that shows it, since it will never happen.

cc @johndoates @edwardhorsford

Reply:
Summary page content with no dates that is now potentially not needed:

<img width="705" alt="Screenshot 2023-03-28 at 10 09 14" src="https://user-images.githubusercontent.com/18436946/228188693-ad5a6806-89d3-4b0d-a767-52cd7a8512be.png">

Reply:
sounds good! let's delete the partial and remove

```
<% elsif @error_rows_count == @total_rows_count %>
  <%= render("summary_all_errors") %>
```

from `show.html.erb`.